radicle-node/src/service/peer.rs

```diff
@@ -89,4 +111,20 @@ impl Session {
     pub fn connected(&mut self, _link: Link) {
         self.attempts = 0;
     }
+
+    pub fn ping(&mut self, reactor: &mut Reactor, rng: &Rng) -> Result<(), SessionError> {
+        if let SessionState::Negotiated { ping, .. } = &mut self.state {
+            let ponglen = rng.u16(0..MAX_MESSAGE_SIZE_BYTES);
+            let msg = Message::Ping {
+                ponglen,
+                zeroes: message::ZeroBytes(
+                    rng.u16(0..(MAX_MESSAGE_SIZE_BYTES - (size_of::<u16>() as u16))),
+                ),
+            };
```

Comment:
@cloudhead how do you feel about this approach?

Instead of having a PING_SIZE constant (and maybe PONG_SIZE), define the max message size as a single constant.  The intent is communicated clearly within the logic now.

Reply:
Yes, I like the approach, however you forgot one small detail: we need to account for both the message type (2 bytes) and the ponglen (2 bytes). Would be worth adding a test with the max size, to make sure the entire message once encoded is exactly `2^16 - 1`

I would also not expose the internals of `ZeroBytes`, but rather have a constructor, eg. `ZeroBytes::new(size: Size)`.

Reply:
Ah and it might make sense for readability to use `size_of::<wire::Size>` to be clearer on intent.